core-bundle/src/Translation/Translator.php

```diff
@@ -64,6 +65,12 @@ public function trans($id, array $parameters = [], $domain = null, $locale = nul
             $translated = vsprintf($translated, $parameters);
         }
 
+        // Restore previous translations in $GLOBALS['TL_LANG'] (#5371)
+        if (null !== $locale && $locale !== $this->getLocale()) {
+            $system = $this->framework->getAdapter(System::class);
+            $system->loadLanguageFile(substr($domain, 7), $this->getLocale());
```

Comment:
Why are you not using `$this->getCatalogue($locale)` to restore the translations?

Reply:
That won't restore the translations. The catalogue calls `System::loadLanguageFile` only when actually retrieving a translation.

Reply:
But now the translator knows how to load language files which previously only the catalogue had to know. This violates the "separation of concerns" principle. @ausi /cc

Reply:
I think our translator and catalogue class are strongly connected, so I don’t see a problem with this here. The `contao_` prefix check for example is also used in both classes.

Reply:
If we want to separate it, we could introduce a `public function loadLanguageFile` in our `MessageCatalogue`.

Reply:
Why not simply use `$this->getCatalogue($this->getLocale())->get($id, $domain);`? This should do the job and it will not break the separation of concerns principle.

Reply:
I opted to not do it this way, because `get` will do more than just `System::loadLanguageFile`. Specifically this would be executed every time, even though it's not necessary:

https://github.com/contao/contao/blob/f81d9e91c36d0d065ae14e0c2703d15c837560fb/core-bundle/src/Translation/MessageCatalogue.php#L164-L186